Remove mining from eth_ namespace

https://github.com/ethereum/execution-apis/pull/430

Trying to get my env working.  Getting this...`error MSB4185`
![image](https://github.com/NethermindEth/nethermind/assets/8411406/9d32835e-b2af-49bb-b145-7ca468f2d5b5)

--- a/src/Nethermind/Nethermind.JsonRpc.Test/JsonRpcServiceTests.cs
+++ b/src/Nethermind/Nethermind.JsonRpc.Test/JsonRpcServiceTests.cs
@@ -151,17 +151,6 @@ namespace Nethermind.JsonRpc.Test
         }
 
         [Test]
-        public void GetWorkTest()
-        {
-            IEthRpcModule ethRpcModule = Substitute.For<IEthRpcModule>();
-            ethRpcModule.eth_getWork().ReturnsForAnyArgs(x => ResultWrapper<IEnumerable<byte[]>>.Success(new[] { Bytes.FromHexString("aa"), Bytes.FromHexString("01") }));
-            JsonRpcSuccessResponse? response = TestRequest(ethRpcModule, "eth_getWork") as JsonRpcSuccessResponse;
-            byte[][]? dataList = response?.Result as byte[][];
-            Assert.NotNull(dataList?.SingleOrDefault(d => d.ToHexString(true) == "0xaa"));
-            Assert.NotNull(dataList?.SingleOrDefault(d => d.ToHexString(true) == "0x01"));
-        }
-
-        [Test]
         public void IncorrectMethodNameTest()
         {
             JsonRpcErrorResponse? response = TestRequest(Substitute.For<IEthRpcModule>(), "incorrect_method") as JsonRpcErrorResponse;

--- a/src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.cs
+++ b/src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.cs
@@ -705,30 +705,6 @@ public partial class EthRpcModuleTests
     }
 
     [Test]
-    public async Task Eth_mining_true()
-    {
-        using Context ctx = await Context.Create();
-        IBlockchainBridge bridge = Substitute.For<IBlockchainBridge>();
-        bridge.IsMining.Returns(true);
-        ctx.Test = await TestRpcBlockchain.ForTest(SealEngineType.NethDev).WithBlockchainBridge(bridge).Build();
-
-        string serialized = await ctx.Test.TestEthRpc("eth_mining");
-        Assert.That(serialized, Is.EqualTo("{\"jsonrpc\":\"2.0\",\"result\":true,\"id\":67}"));
-    }
-
-    [Test]
-    public async Task Eth_mining_false()
-    {
-        using Context ctx = await Context.Create();
-        IBlockchainBridge bridge = Substitute.For<IBlockchainBridge>();
-        bridge.IsMining.Returns(false);
-        ctx.Test = await TestRpcBlockchain.ForTest(SealEngineType.NethDev).WithBlockchainBridge(bridge).Build();
-
-        string serialized = await ctx.Test.TestEthRpc("eth_mining");
-        Assert.That(serialized, Is.EqualTo("{\"jsonrpc\":\"2.0\",\"result\":false,\"id\":67}"));
-    }
-
-    [Test]
     public async Task Eth_accounts()
     {
         using Context ctx = await Context.Create();

--- a/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs
+++ b/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs
@@ -110,16 +110,6 @@ public partial class EthRpcModule : IEthRpcModule
         return ResultWrapper<Address>.Success(Address.Zero);
     }
 
-    public ResultWrapper<bool?> eth_mining()
-    {
-        return ResultWrapper<bool?>.Success(_blockchainBridge.IsMining);
-    }
-
-    public ResultWrapper<UInt256?> eth_hashrate()
-    {
-        return ResultWrapper<UInt256?>.Success(0);
-    }
-
     public ResultWrapper<UInt256?> eth_gasPrice()
     {
         return ResultWrapper<UInt256?>.Success(_gasPriceOracle.GetGasPriceEstimate());
@@ -650,21 +640,6 @@ public partial class EthRpcModule : IEthRpcModule
         {
             return GetFailureResult<IEnumerable<FilterLog>>(exception, _ethSyncingInfo.SyncMode.HaveNotSyncedReceiptsYet());
         }
-    }
-
-    public ResultWrapper<IEnumerable<byte[]>> eth_getWork()
-    {
-        return ResultWrapper<IEnumerable<byte[]>>.Fail("eth_getWork not supported", ErrorCodes.MethodNotFound);
-    }
-
-    public ResultWrapper<bool?> eth_submitWork(byte[] nonce, Keccak headerPowHash, byte[] mixDigest)
-    {
-        return ResultWrapper<bool?>.Fail("eth_submitWork not supported", ErrorCodes.MethodNotFound, null);
-    }
-
-    public ResultWrapper<bool?> eth_submitHashrate(string hashRate, string id)
-    {
-        return ResultWrapper<bool?>.Fail("eth_submitHashrate not supported", ErrorCodes.MethodNotFound, null);
     }
 
     // https://github.com/ethereum/EIPs/issues/1186

--- a/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/IEthRpcModule.cs
+++ b/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/IEthRpcModule.cs
@@ -41,9 +41,6 @@ namespace Nethermind.JsonRpc.Modules.Eth
             ExampleResponse = "0x0000000000000000000000000000000000000000")]
         ResultWrapper<Address> eth_coinbase();
 
-        [JsonRpcMethod(IsImplemented = false, Description = "Returns mining status", IsSharable = true)]
-        ResultWrapper<bool?> eth_mining();
-
         [JsonRpcMethod(IsImplemented = true,
             Description = "Returns block fee history.",
             IsSharable = true,
@@ -55,12 +52,6 @@ namespace Nethermind.JsonRpc.Modules.Eth
 
         [JsonRpcMethod(IsImplemented = false, Description = "", IsSharable = true)]
         ResultWrapper<UInt256?> eth_maxPriorityFeePerGas();
-
-        [JsonRpcMethod(IsImplemented = false,
-            Description = "Returns mining hashrate",
-            IsSharable = true,
-            ExampleResponse = "0x0")]
-        ResultWrapper<UInt256?> eth_hashrate();
 
         [JsonRpcMethod(IsImplemented = false,
             Description = "Returns miner's gas price",
@@ -257,15 +248,6 @@ namespace Nethermind.JsonRpc.Modules.Eth
 
         [JsonRpcMethod(IsImplemented = true, Description = "Reads logs", IsSharable = false)]
         ResultWrapper<IEnumerable<FilterLog>> eth_getLogs(Filter filter);
-
-        [JsonRpcMethod(Description = "", IsImplemented = false, IsSharable = true)]
-        ResultWrapper<IEnumerable<byte[]>> eth_getWork();
-
-        [JsonRpcMethod(Description = "", IsImplemented = false, IsSharable = false)]
-        ResultWrapper<bool?> eth_submitWork(byte[] nonce, Keccak headerPowHash, byte[] mixDigest);
-
-        [JsonRpcMethod(Description = "", IsImplemented = false, IsSharable = false)]
-        ResultWrapper<bool?> eth_submitHashrate(string hashRate, string id);
 
         [JsonRpcMethod(Description = "https://github.com/ethereum/EIPs/issues/1186",
             IsImplemented = true,

